src/bubble_swarm/bubble_circle.rs

```diff
@@ -0,0 +1,74 @@
+use crate::math2d::{circle::Circle, point::Point, vector::Vector};
+
+use super::{
+    source_circle::SourceCircle,
+    utils::{calculate_circle_intersection, get_mid_point},
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct BubbleCircle {
+    circle: Circle,
+    source_circle: SourceCircle,
+}
+
+impl BubbleCircle {
+    pub fn new(circle: Circle, source_circle: SourceCircle) -> Self {
+        Self {
+            circle,
+            source_circle,
+        }
+    }
+
+    fn get_radius(&self, center: Point) -> usize {
```

Comment:
To go with your sentiment "We should use float". I would introduce a return type f32 here and have the cast explicitly at the call site and not implicitly here. Naming it get_positive_radius might also be clearer.